src/legacy/core_plugins/visualizations/public/np_ready/types/types_service.ts

```diff
@@ -17,41 +17,59 @@
  * under the License.
  */
 
+import { IconType } from '@elastic/eui';
 import { visTypeAliasRegistry, VisTypeAlias } from './vis_type_alias_registry';
 
-interface SetupDependencies {
-  Vis: any;
-  VisFactoryProvider: any;
-  VisTypesRegistryProvider: any;
-}
+export interface VisType {
+  name: string;
+  title: string;
+  description?: string;
+  visualization: any;
+  isAccessible?: boolean;
+  requestHandler: string;
+  responseHandler: string;
+  icon?: IconType;
+  image?: string;
+  stage: 'experimental' | 'production';
+  requiresSearch: boolean;
+  hidden: boolean;
 
-interface StartDependencies {
-  VisTypesRegistryProvider: any;
-  Private: any;
+  // Since we haven't typed everything here yet, we basically "any" the rest
+  // of that interface. This should be removed as soon as this type definition
+  // has been completed. But that way we at least have typing for a couple of
+  // properties on that type.
+  [key: string]: any;
```

Comment:
I would suggest to try removing this line, because what this line does, it allows to assign absolutely anything to absolutely any key of `VisType`.

Instead, if we removed this line, any user of this interface would be forced to go here and update this interface (that's what we want), or deal with it by manually by casting their variables.

Reply:
lets do this before in the next PR where we'll be fixing visualizations plugin to export only whats really needed (right now it exports too much)